asynciterator.ts

```diff
@@ -25,6 +25,48 @@ export function setTaskScheduler(scheduler: TaskScheduler): void {
   taskScheduler = scheduler;
 }
 
+// Determines whether the given object is a function
+function isFunction(object: any): object is Function {
+  return typeof object === 'function';
+}
+
+// Determines whether the given object is an EventEmitter
+function isEventEmitter(object: any): object is EventEmitter {
+  return object && typeof object.on === 'function';
+}
+
+// Determines whether the given object is a promise
+function isPromise<T>(object: any): object is Promise<T> {
+  return object && typeof object.then === 'function';
+}
+
+// Determines whether the given object is a source expression
+function isSourceExpression<T>(object: any): object is SourceExpression<T> {
+  return object && (isEventEmitter(object) || isPromise(object) || isFunction(object));
+}
+
+// Determines whether the given object implements basic streaming
+// interfaces common to both stream.Readable and AsyncIterator
+/* eslint-disable arrow-body-style */
+export const isAsyncIteratorLike = <T>(item: { [key: string]: any }): item is AsyncIteratorLike<T> => {
+  return isFunction(item.on) && isFunction(item.read);
+};
+
+// Determines whether the given object is an instance of ES Iterator
+export const isIterator = <T>(item: { [key: string]: any }): item is Iterator<T> => {
+  return isFunction(item.next);
+};
+
+// Determines whether the given object is an instance of ES Iterable
+export const isIterable = <T>(item: { [key: string]: any }): item is Iterable<T> => {
+  return Symbol.iterator in item;
+};
+
+// Determines whether the given object is an instance of ES Iterable or an instance of ES Iterator
+export const isIterableOrIterator = <T>(item: { [key: string]: any }): item is Iterable<T> | Iterator<T> => {
+  return isIterable<T>(item) || isIterator<T>(item);
+};
+
```

Comment:
Why are you using arrow-body style here?

Reply:
Habit - I tend to favor arrow functions over normal functions. That said, other `is*` functions are normal functions so I changed the new ones into normal functions, too.